operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -332,12 +308,84 @@ private static boolean isCurrentPodValid(V1Pod build, V1Pod current) {
+      V1Container build, V1Container current, List<String> ignoring) {
+    return current.getImage().equals(build.getImage())
+        && current.getImagePullPolicy().equals(build.getImagePullPolicy())
+        && Objects.equals(current.getSecurityContext(), build.getSecurityContext())
+        && equalSettings(current.getLivenessProbe(), build.getLivenessProbe())
+        && equalSettings(current.getReadinessProbe(), build.getReadinessProbe())
+        && equalSets(mountsWithout(current.getVolumeMounts(), ignoring), build.getVolumeMounts())
+        && equalSets(current.getPorts(), build.getPorts())
+        && equalSets(current.getEnv(), build.getEnv())
+        && equalSets(current.getEnvFrom(), build.getEnvFrom());
+  }
+
+  private static boolean equalSettings(V1Probe probe1, V1Probe probe2) {
+    return Objects.equals(probe1.getInitialDelaySeconds(), probe2.getInitialDelaySeconds())
+        && Objects.equals(probe1.getTimeoutSeconds(), probe2.getTimeoutSeconds())
+        && Objects.equals(probe1.getPeriodSeconds(), probe2.getPeriodSeconds());
+  }
+
+  private static List<V1Volume> volumesWithout(
+      List<V1Volume> volumeMounts, List<String> volumesToIgnore) {
+    List<V1Volume> result = new ArrayList<>(volumeMounts);
+    for (Iterator<V1Volume> each = result.iterator(); each.hasNext(); )
+      if (volumesToIgnore.contains(each.next().getName())) each.remove();
+
+    return result;
+  }
+
+  private static List<V1VolumeMount> mountsWithout(
+      List<V1VolumeMount> volumeMounts, List<String> volumesToIgnore) {
+    List<V1VolumeMount> result = new ArrayList<>(volumeMounts);
+    for (Iterator<V1VolumeMount> each = result.iterator(); each.hasNext(); )
+      if (volumesToIgnore.contains(each.next().getName())) each.remove();
+
+    return result;
+  }
+
+  private static List<String> getVolumesToIgnore(V1Pod current) {
+    List<String> k8sVolumeNames = new ArrayList<>();
+    for (V1Container container : getContainers(current))
+      for (V1VolumeMount mount : getVolumeMounts(container))
+        if (PodDefaults.K8S_SERVICE_ACCOUNT_MOUNT_PATH.equals(mount.getMountPath()))
+          k8sVolumeNames.add(mount.getName());
+
+    return k8sVolumeNames;
+  }
+
+  private static List<V1Container> getContainers(V1Pod current) {
+    return Optional.ofNullable(current.getSpec().getContainers()).orElse(Collections.emptyList());
+  }
+
+  private static List<V1VolumeMount> getVolumeMounts(V1Container container) {
+    return Optional.ofNullable(container.getVolumeMounts()).orElse(Collections.emptyList());
+  }
+
+  private static Map<String, String> getCustomerLabels(V1ObjectMeta metadata) {
+    Map<String, String> result = new HashMap<>();
+    for (Map.Entry<String, String> entry : metadata.getLabels().entrySet())
+      if (!isOperatorLabel(entry)) result.put(entry.getKey(), entry.getValue());
+    return result;
+  }
+
+  private static boolean isOperatorLabel(Map.Entry<String, String> label) {
```

Comment:
I don't think we tell customers they can't add labels that start with 'weblogic.'

Is there a way you could rework this to tell it to ignore the ones we know that k8s adds instead?

Reply:
How about adding a label to the server pod that lists the names of the labels the user added (since this code has access to the domain resource).

Then you could use that label to find the names of the labels to compare when deciding if a pod is valid.

Reply:
Russ said that Ryan is OK with telling customers that they're not allowed to add labels that start with 'weblogic.'.

Do/should we check for this anywhere?

And where is this restriction documented?